calculation_scripts/OptimizationWithWrapper.py

```diff
@@ -0,0 +1,92 @@
+import dda_model
+from dda_model import optimizers
+import numpy as np
+import matplotlib.pyplot as plt
+import os
+
+def construct_model():
+    sym_axis = [10.5, 10.5]
+    geometry_shape = [22, 22, 10]
+    pixel_size = 15.0
+    light_direction = [0, 0, 1]
+    light_polarization = [1, 0, 0]
+    wavelength = 542
+    initialization = np.loadtxt("initializations/hourglass.txt")
+    dielectric_constants = [1.01 + 0j, 5.96282 + 3.80423e-7j]
+
+    model = dda_model.DDAModelWrapper(
+        geometry_shape,
+        pixel_size,
+        initialization,
+        dielectric_constants=dielectric_constants,
+        light_direction=light_direction,
+        light_polarization=light_polarization,
+        light_wavelength_nm=wavelength,
+        symmetry_axes=sym_axis,
+        verbose=True,
+    )
+    return model
+
+def saveCurrentStructure(model_, path, iteration):
+    all_parameters = model_.allParameters()
+
+    with open(path + '\\CoreStructure\\CoreStructure' + str(iteration) + '.txt', 'w') as f:
+        for para in all_parameters:
+            f.write(f"{para}\n")
+
+def plotObjectiveFunction(objective_values, path):
+    plt.figure(1)
+    plt.plot(objective_values)
+    #plt.legend(loc='lower right')
+    plt.title('Objective Function Plot')
+    plt.ylabel('Objective Function')
+    plt.xlabel('Iteration #')
+    plt.rc('axes', titlesize=14)     # fontsize of the axes title
+    plt.rc('axes', labelsize=12)    # fontsize of the x and y labels
+    plt.savefig(path + '\\obj.png', bbox_inches='tight')
+
+model = construct_model()
+evo_max_iter = 400
+# epsilon = 0.01 # This works better
+epsilon = 0.01
+all_objective_values = [0] * evo_max_iter
```

Comment:
You can leave this as `all_objective_values = []` and then call `all_objective_values.append(value)`. This is less likely to throw an error at the end of your code, since you are no longer indexing into `all_objective_values` and therefore can no longer through an IndexError

Reply:
done - good idea!